components/health-check-proxy/cmd/health-check/main.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"log"
+)
+
+func main() {
+	var (
+		path       string
+		host       string
+		statusPort int
+		retries    int
+	)
+	flag.StringVar(&path, "path", "", "liveness or readiness endpoint")
```

Comment:
just accept only one flag e.g. called URL. It's simple and generic. When you split that with different flag like host, port, path and then building your final url is more complicated and end-users do not have a full control e.g. `http` protocol is right now hardcoded